cub/cub/device/dispatch/dispatch_rle.cuh

```diff
@@ -302,152 +392,166 @@ struct DeviceRleDispatch
 
     constexpr int block_threads    = ActivePolicyT::RleSweepPolicyT::BLOCK_THREADS;
     constexpr int items_per_thread = ActivePolicyT::RleSweepPolicyT::ITEMS_PER_THREAD;
+    constexpr auto tile_size       = static_cast<global_offset_t>(block_threads * items_per_thread);
 
-    do
+    // The upper bound of for the number of items that a single kernel invocation will ever process
+    auto capped_num_items_per_invocation = num_items;
+    if constexpr (use_streaming_invocation)
     {
-      // Get device ordinal
-      int device_ordinal;
-      error = CubDebug(cudaGetDevice(&device_ordinal));
-      if (cudaSuccess != error)
-      {
-        break;
-      }
+      capped_num_items_per_invocation = static_cast<global_offset_t>(_CUDA_VSTD::numeric_limits<local_offset_t>::max());
+      // Make sure that the number of items is a multiple of tile size
+      capped_num_items_per_invocation -= (capped_num_items_per_invocation % tile_size);
+    }
+
+    // Across invocations, the maximum number of items that a single kernel invocation will ever process
+    const auto max_num_items_per_invocation =
+      use_streaming_invocation ? _CUDA_VSTD::min(capped_num_items_per_invocation, num_items) : num_items;
+
+    // Number of invocations required to "iterate" over the total input (at least one iteration to process zero items)
+    auto const num_partitions =
+      (num_items == 0) ? global_offset_t{1} : ::cuda::ceil_div(num_items, capped_num_items_per_invocation);
```

Comment:
Suggestion: it should not change the outcome, but this would be easier to understand for me, since we typically have to check the divisor for zero:
```suggestion
    auto const num_partitions =
      (capped_num_items_per_invocation == 0) ? global_offset_t{1} : ::cuda::ceil_div(num_items, capped_num_items_per_invocation);
```